src/components/views/rooms/PinnedEventsPanel.js

```diff
@@ -0,0 +1,171 @@
+import { _t } from "matrix-react-sdk/lib/languageHandler";
+import { EventTimeline } from "matrix-js-sdk";
+
+const PinnedEventTile = React.createClass({
+    displayName: 'PinnedEventTile',
+    propTypes: {
+        mxRoom: React.PropTypes.object.isRequired,
+        mxEvent: React.PropTypes.object.isRequired,
+        onUnpinned: React.PropTypes.func,
+    },
+    onTileClicked: function() {
+        dis.dispatch({
+            action: 'view_room',
+            event_id: this.props.mxEvent.getId(),
+            highlighted: true,
+            room_id: this.props.mxEvent.getRoomId(),
+        });
+    },
+    onUnpinClicked: function() {
+        const pinnedEvents = this.props.mxRoom.currentState.getStateEvents("m.room.pinned_events", "");
+        if (!pinnedEvents || !pinnedEvents.getContent().pinned) {
+            // Nothing to do: already unpinned
+            if (this.props.onUnpinned) this.props.onUnpinned();
+        } else {
+            const pinned = pinnedEvents.getContent().pinned;
+            const index = pinned.indexOf(this.props.mxEvent.getId());
+            if (index !== -1) {
+                pinned.splice(index, 1);
+                MatrixClientPeg.get().sendStateEvent(this.props.mxRoom.roomId, 'm.room.pinned_events', {pinned}, '').then(() => {
+                    if (this.props.onUnpinned) this.props.onUnpinned();
+                });
+            } else if (this.props.onUnpinned) this.props.onUnpinned();
+        }
+    },
+    _canUnpin: function() {
+        return this.props.mxRoom.currentState.mayClientSendStateEvent('m.room.pinned_events', MatrixClientPeg.get());
+    },
+    render: function() {
+        const MessageEvent = sdk.getComponent("views.messages.MessageEvent");
+        const MemberAvatar = sdk.getComponent("views.avatars.MemberAvatar");
+
+        const sender = this.props.mxRoom.getMember(this.props.mxEvent.getSender());
+        const avatarSize = 40;
+
+        let unpinButton = null;
+        if (this._canUnpin()) {
+            unpinButton = <img src="img/cancel-red.svg" className="mx_PinnedEventTile_unpinButton" width="8" height="8"
+                               onClick={this.onUnpinClicked} alt={_t('Unpin Message')} title={_t('Unpin Message')} />;
+        }
+
+        return (
+            <div className="mx_PinnedEventTile">
+                <div className="mx_PinnedEventTile_actions">
+                    <AccessibleButton className="mx_PinnedEventTile_gotoButton mx_textButton" onClick={this.onTileClicked}>
+                        { _t("Jump to message") }
+                    </AccessibleButton>
+                    { unpinButton }
+                </div>
+
+                <MemberAvatar member={sender} width={avatarSize} height={avatarSize} />
+                <span className="mx_PinnedEventTile_sender">
+                    {sender.name}
+                </span>
+                <MessageEvent mxEvent={this.props.mxEvent} className="mx_PinnedEventTile_body" />
```

Comment:
I'm surprised we can't re-use a class from regular Event Tiles here - duplicating the layout seems odd.

Reply:
Not using event tiles is intentional to have a slightly different layout. Event tiles have excessive whitespace, timestamps, and support read receipts. This layout is a bit more compact, has a larger avatar, and has other buttons in places.